### PR TITLE
Adding metadata to information when calling chain.execute(string)

### PR DESCRIPTION
**Pull request by mateusscheper**
The goal of this change is to allow the user to add extra information when appending the retrieved docs informations on chain.execute(string). 
Example: 

```
chain = ConversationalRetrievalChain.builder()               
                .metadata(List.of("file_name"))
                [...]
                .build();
String response = chain.execute("My question");
```
When building the prompt, it will include the information like: 
```
information -> ...file_name: my_file.pdf
Content: my first retrieved content....

...file_name: my_file.pdf
Content: my second retrieved content...
```

Since it's my first contribution here, please let me know if something's wrong. Thanks!

### CODE REVIEW

File: langchain4j/src/main/java/dev/langchain4j/chain/ConversationalRetrievalChain.java [hunk 1]
```diff
@@ -36,15 +36,19 @@ public class ConversationalRetrievalChain implements Chain<String, String> {
...
 
+    private final List<String> metadata;
+
     @Builder
     public ConversationalRetrievalChain(ChatLanguageModel chatLanguageModel,
                                         ChatMemory chatMemory,
                                         PromptTemplate promptTemplate,
-                                        Retriever<TextSegment> retriever) {
+                                        Retriever<TextSegment> retriever,
+                                        List<String> metadata) {
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/chain/ConversationalRetrievalChain.java]: I would name it a bit more specific, something like `metadataKeysToInclude` or `includedMetadataKeys`

File: langchain4j/src/main/java/dev/langchain4j/chain/ConversationalRetrievalChain.java [hunk 2]
```diff
@@ -69,10 +73,33 @@ public String execute(String question) {
...
+        }
+
+        if (!formattedText.toString().isEmpty()) {
+            formattedText.append("Content: ");
+        }
+
+        formattedText.append(textSegment.text());
+
+        return formattedText.toString();
+    }
```
Comment (langchain4j) [inline langchain4j/src/main/java/dev/langchain4j/chain/ConversationalRetrievalChain.java:104]: could you please add a few unit tests to make sure this togic works as expected? e.g. without specified metadata keys to include, with 1,2 keys, with 0, 1, 2 segments, etc

**Review by ghost — COMMENTED** [review]
@mateusscheper thanks a lot for your contribution! Please check my comments.

**Review by ghost — APPROVED** [review]
@mateusscheper looks great, thank you!

**Pull request merged**